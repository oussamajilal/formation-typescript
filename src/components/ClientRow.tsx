import { FC, useState } from "react";
import ClientNotif from "./ClientNotif";

interface ClientRowProps {
  id: number;
  firstName: string;
  lastName: string;
}

const ClientRow: FC<ClientRowProps> = (props) => {
  const [inc, setInc] = useState(0);
  const [decrement, setDecrement] = useState(0);

  const handleIncClick = () => {
    setInc(inc + 1);
  };
  const handleDecrementClick = () => {
    setDecrement(decrement + 1);
  };
  return (
    <tr>
      <td>
        <ClientNotif inc={inc} decrement={decrement} />
      </td>
      <td>{props.id}</td>
      <td>{props.lastName}</td>
      <td>Last Name: {props.lastName}</td>
      <td data-testid="inc-cell">{inc}</td>
      <td data-testid="dec-cell">{decrement}</td>
      <td>
        <button data-testid="inc-btn" onClick={handleIncClick}>
          Increment
        </button>
        <button data-testid="dec-btn" onClick={handleDecrementClick}>
          Decrement
        </button>
      </td>
    </tr>
  );
};

export default ClientRow;
