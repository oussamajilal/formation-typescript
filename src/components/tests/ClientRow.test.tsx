import { render, RenderResult } from "@testing-library/react";
import ClientRow from "../ClientRow";

let dom: RenderResult;

beforeEach(() => {
  dom = render(
    <ClientRow
      id={1}
      firstName={"fake first name"}
      lastName={"fake last name"}
    />
  );
});

describe("ClientRow", () => {
  it("should render correctly", () => {
    expect(dom.asFragment()).toMatchSnapshot();
  });

  it("should increment the cell when I click on the inc button", () => {
    expect(dom.getByTestId("inc-cell")).toHaveTextContent("0");

    dom.getByTestId("inc-btn").click();

    expect(dom.getByTestId("inc-cell")).toHaveTextContent("1");

    dom.getByTestId("inc-btn").click();

    expect(dom.getByTestId("inc-cell")).toHaveTextContent("2");
  });

  it("should decrement the cell when I click on the inc button", () => {
    expect(dom.getByTestId("dec-cell")).toHaveTextContent("0");

    dom.getByTestId("dec-btn").click();

    expect(dom.getByTestId("dec-cell")).toHaveTextContent("1");

    dom.getByTestId("dec-btn").click();

    expect(dom.getByTestId("dec-cell")).toHaveTextContent("2");
  });
});
