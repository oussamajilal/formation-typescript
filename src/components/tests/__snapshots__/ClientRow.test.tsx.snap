// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientRow should render correctly 1`] = `
<DocumentFragment>
  1fake last nameLast Name: fake last name00
  <button
    data-testid="inc-btn"
  >
    Increment
  </button>
  <button
    data-testid="dec-btn"
  >
    Decrement
  </button>
</DocumentFragment>
`;
